src/core/Akka.Persistence/Journal/AsyncRecovery.cs

```diff
@@ -62,7 +63,8 @@ public interface IAsyncRecovery
         /// <param name="fromSequenceNr">Hint where to start searching for the highest sequence number.
         /// When a persistent actor is recovering this <paramref name="fromSequenceNr"/> will the sequence
         /// number of the used snapshot, or `0L` if no snapshot is used.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/> used to signal cancelled recovery operation</param>
         /// <returns>TBD</returns>
-        Task<long> ReadHighestSequenceNrAsync(string persistenceId, long fromSequenceNr);
+        Task<long> ReadHighestSequenceNrAsync(string persistenceId, long fromSequenceNr, CancellationToken cancellationToken);
```

Comment:
[Breaking change] `Akka.Persistence` codes that are being wrapped inside a `CircuitBreaker` will now take a `CancellationToken` as a signal that the operation has been timed out by the `CircuitBreaker`.

Reply:
LGTM